contents/projectManagement/testDrivenDevelopment.md

```diff
@@ -0,0 +1,199 @@
+
+Figure 1 below shows the general TDD cycle.
+
+<center>
+<pic src="images/tddSteps.jpg" alt="tddSteps" height="420">
+
+_Figure 1. General TDD cycle._[^2]
+</pic>
+</center>
+
+The essence of TDD can therefore be summarized into the mnemonic "Red, Green, Refactor", where "Red" signifies failure and "Green" signifies success (of the test). "Refactor" refers to the refactoring step in the TDD cycle.
+
+<center>
+<pic src="images/mantra.jpg" alt="mantra" height="170">
+
+_Figure 2. Red, Green, Refactor._[^2]
+</pic>
+</center>
+
+## Why adopt Test-Driven Development?
+
+Some reasons why teams can choose to adopt the approach of test-driven development are: 
+
+### Benefit 1: Improves Code Quality
+
+TDD can lead to well-written code. This is a direct consequence of the cycle that was explained [above](#development-cycle). One of the best practices while following TDD is to focus on small units of code. This can lead to more modular, flexible code with looser coupling that can be scaled[^3]. This way the code unit targets a specific requirement or feature with just enough code to fail, and descriptive enough for a new-comer to understand it. Additionally, developers will work on units that can be written and tested independently and integrated later. This can be achieved through <trigger trigger="click" for="modal:index-mock">mock frameworks</trigger>, which help drive home the rationale of writing unit tests.
+
+The tests also cover all possible branches and paths that the production code can take because no more code is written than is necessary[^3]. Detecting bugs at early stages is easier when you're working with code that has high coverage [^5]. 
+
+<modal title="Mock Objects and Frameworks" id="modal:index-mock">
+Mock objects are simulated objects whose sole purpose is to act as a black-box that mimics the behaviour of real-world objects during testing. They are especially useful in writing unit tests, which are meant to test the functionality of a portion of code assuming that the rest of the code behaves as it should. 
+</modal>
+
+### Benefit 2: Ensures that Product Requirements are Understood
+
+Test-driven development requires the developers to have a good understanding of the product or feature requirements before any sort of development can begin. As mentioned earlier, the first step in development is (ideally) writing tests. Coming up with tests without a clear picture of the requirements or specifications  can be dangerous as - 
+
+1. you could end up testing something that doesn't need to be tested, thus giving you a false sense of accomplishment when you write production code to pass the test, or 
+1. you could end up overlooking some requirements that _must_ be tested, which is far worse.
+
+TDD thus prevents the team from rushing into development, and pushes them to prioritize the requirements-gathering step. 
```

Comment:
This point essentially means that development teams will be clear with any and every requirement. This is beneficial because there are a lot of instances where teams (irrespective of whether they follow TDD or not) rush into development without a clear understanding of what's required and waste time in the future trying to iron out the requirements. 
It shows that teams that are confident about the requirements will not face the issues listed above.

Reply:
Ahh, did you mean that prioritising functional code over tests first has a higher chance of resulting in writing unnecessary tests?

I think I misunderstood this because of this line
`As mentioned earlier, the first step in development is (ideally) writing tests. Coming up with tests without a clear picture of the requirements or specifications can be dangerous as -`

It seems like the reasons why it could be dangerous are being attributed to TDD because of the earlier sentence. 

Perhaps you could try something like `As mentioned, ... Hence, Test-driven development requires the developers...`, ( just a restructure of your paragraph )
then `When tests are written after functional code, there is a higher chance that requirements are ..., which can be dangerous since...`